lib/ical/recur_iterator.js

```diff
@@ -409,6 +421,49 @@ ICAL.RecurIterator = (function() {
       this.increment_generic(inc, "hour", 24, "monthday");
     },
 
+    get_max_number_of_possible_byday_values: function get_number_of_possible_byday_values(days_in_timeslice) {
+      var self = this;
+      var reducer = function(acc, value) {
+        var whatNumberOfDay = self.ruleDayOfWeek(value)[0];
+        if (whatNumberOfDay === 0) {
+          return acc + days_in_timeslice;
+        }
+        return acc + 1
+      };
+      return this.by_data.BYDAY.reduce(reducer, 0);
+    },
+
+    does_any_value_exceed_max_days: function does_any_value_exceed_max_days(values, max_number_of_possible_days) {
+      for (var value of values) {
+        if (value > max_number_of_possible_days) {
+          return true;
+        }
+      }
+      return false;
+    },
+
+    has_invalid_bysetpos: function has_invalid_bysetpos(days_in_timeslice) {
+      // days_in_timeslice math:
+      // there are at most 5 of any given day in a month, and 53 of any given day in a year
```

Comment:
Might be worth duplicating the bit about 5 days in a month where the `5` appears above to keep it from being mistaken for a magic number.